test/async-await.test.js

```diff
@@ -423,6 +423,28 @@ test('promise was fulfilled with undefined', t => {
   })
 })
 
+test('promise was fulfilled with undefined using inject', async (t) => {
+  const stream = split(JSON.parse)
+  const fastify = Fastify({
+    logger: {
+      stream,
+      level: 'error'
+    }
+  })
+
+  fastify.get('/', async (req, reply) => {
+  })
+
+  stream.once('data', line => {
+    t.fail('should not log an error')
+  })
+
+  const res = await fastify.inject('/')
+
+  t.equal(res.body, '')
```

Comment:
the body should be null, but I guess it's another edge case due to mimicking built-in Request/Response in light-my-request

Reply:
There is no such thing as a null http body, it's either a string or buffer of lenght 0.

Reply:
well if you look at the test in the same file with a plan http client, you can see that it asserts against undefined https://github.com/fastify/fastify/pull/4211/files#diff-1c26a7df32f7d09a699227287432ba17f6a97376fd6e3e6f334256b2f5b4c526R420

Reply:
That's how sget is implementing it, I would argue it's not really correct.